plugins/inputs/intel_pmt/README.md

```diff
@@ -0,0 +1,434 @@
+  # datatype_metrics = []
+  
+  ## Filter metrics by their name.
+  ## See the Filtering section in README for more details.
+  ## If empty, all metrics are collected.
+  # sample_metrics = []
+```
+
+## Example Configuration: `pmt_source` as a filepath
+
+This configuration uses a local copy of the XML specification:
+
+```toml
+[[inputs.intel_pmt]]
+  pmt_source = "/home/telegraf/Intel-PMT/xml/pmt.xml"
+```
+
+## Example Configuration: C-State residency and temperature with a datatype metric filter
+
+This configuration allows getting only a subset of metrics
+with the use of a datatype filter:
+
+```toml
+[[inputs.intel_pmt]]
+  pmt_source = "/home/telegraf/Intel-PMT/xml/pmt.xml"
+  datatype_metrics = ["tbandwidth_28b","ttemperature"]
+```
+
+## Example Configuration: C-State residency and temperature with a sample metric filter
+
+This configuration allows getting only a subset of metrics
+with the use of a sample filter:
+
+```toml
+[[inputs.intel_pmt]]
+  pmt_source = "/home/telegraf/Intel-PMT/xml/pmt.xml"
+  sample_metrics = ["C0Residency","C1Residency", "Cx_TEMP"]
+```
+
+## Prerequisites
+
+Minimum Linux kernel version 5.11 with
+
+- the `intel_pmt_telemetry` module loaded (on kernels 5.11-5.14)
+- the `intel_pmt` module loaded (on kernels 5.14+)
+
+Intel PMT is exposed on a limited number of devices, e.g.
+
+- 4th Generation Intel® Xeon® Scalable Processors
+(codenamed Sapphire Rapids / SPR)
+- 6th Generation Intel® Xeon® Scalable Processors
+(codenamed Granite Rapids / GNR)
+
+PMT space is located in `/sys/class/intel_pmt` with `telem` files requiring
+root privileges to read.
+
+### If Telegraf is not running as a root user
+
+By default, the `telem` binary file requires root privileges to be read.
+To avoid running Telegraf as a root, use one of these steps:
+
+- Add read permission for the current user to `telem` files exposed by the
+`intel_pmt` driver
+
+  ```sh
+  sudo chmod o+r /sys/class/intel_pmt/telem*/telem
+  ```
+
+- Add the following capabilities to the Telegraf executable
+
+  ```sh
+  sudo setcap cap_dac_read_search+ep /usr/bin/telegraf
+  ```
+
+## Metrics
+
+All measurements have the following tags:
+
+- `guid` (unique id of an Intel PMT space).
+- `numa_node` (NUMA node the sample is collected from).
+- `sample_name` (name of the gathered sample).
+- `sample_group` (name of a group to which the sample belongs).
+- `datatype_idref` (datatype to which the sample belongs).
+
+Measurements prefixed with `Cx_` where `x` is the core number also have the
+following tag:
+  
+- `core` (core to which the measurement relates).
+  
+Measurements prefixed with `CHAx_` where `x` is the CHA number also have the
+following tag:
+
+- `cha` (Caching and Home Agent to which the measurement relates).
+
+## Filtering
+
+The plugin offers two filtering options for selecting metrics:
+
+- filter by datatype (groups of metrics),
+- filter by name.
+
+It's important to note that only one filtering option
+should be chosen at a time.
```

Comment:
Why do we need special filtering? When the global filter options exist already. I can see possibly the desire for the by datatype, but by name already exists.

Reply:
We need this special filtering to allow the user to filter out unwanted metrics. 

This filtering happens on the XML data, in the initialization of the plugin, so in Gather we aren't filtering anything anymore.

Reply:
> We need this special filtering to allow the user to filter out unwanted metrics.

This is what the global metric filtering is for.

Reply:
With this filtering option the data won't be excluded - it will never be read at all (to be specific: there won't even be a measurement to exclude). It will be filtered out during the XML reading, during the initialization of the plugin.

This saves memory and computational power.

Reply:
Can you quantify this then. This seems a pre-mature optimization

Reply:
Changed this section naming from `Filtering` to `Enabling metrics` to avoid any confusion with global filtering options.

This plugin on 2 socket server currently produces over 12000 metrics every Gather call (as a lot of them are per core/cha). This number will only grow on new devices that support Intel PMT.